### PR TITLE
Created matmul PI for pytorch

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

### CODE REVIEW

File: pytorch/matmul.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import numpy as np
+import json
+from timeit import time
+
+import torch
+
+_LOGGER = logging.getLogger(__name__)
+
+_ARGS_DTYPE_MAP = {
+    "32-bit floating point": {
```
Comment (fridex) [inline pytorch/matmul.py]: Can we have these named as in case of TensorFlow performance indicators? (`float32`, ...)

File: pytorch/matmul.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+def bench(n):
+    if _ARGS_DEVICE == 'gpu':
+        if torch.cuda.is_available():
+            number_GPU = torch.cuda.device_count()
+            curren_device = torch.cuda.current_device()
+            torch.cuda.device(curren_device)
+            name_GPU = torch.cuda.get_device_name(curren_device)
+            device = torch.device('cuda')
+
+        raise Exception("No GPU available")
```
Comment (fridex) [inline pytorch/matmul.py]: Missing  `else`?

File: pytorch/matmul.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    os.environ['CUDA_VISIBLE_DEVICES'] = '-1'
+
+
+def bench(n):
+    if _ARGS_DEVICE == 'gpu':
+        if torch.cuda.is_available():
+            number_GPU = torch.cuda.device_count()
+            curren_device = torch.cuda.current_device()
+            torch.cuda.device(curren_device)
+            name_GPU = torch.cuda.get_device_name(curren_device)
```
Comment (fridex) [inline pytorch/matmul.py]: curren**t**_device (and above).

**Review by fridex — COMMENTED** [review]
Few minor comments, :100: :1st_place_medal: otherwise! :+1:

File: pytorch/matmul.py (hunk 1 above)
Comment (pacospace) [reply pytorch/matmul.py]: PyTorch has more than the one in TensorFlow, so only some of them could be matched.

Comment (fridex) [reply pytorch/matmul.py]: Let's share what they have in common and introduce a new naming where needed. I can imagine a query where we would be comparing TF and PyTorch performance for a specific bit width respecting model used (e.g. optimized models which reduce bit size). IMHO no need to have two names for same thing.

Comment (pacospace) [reply pytorch/matmul.py]: Agreed :) I updated values, matching with TF for the one compatible. WDYT?

Comment (pacospace) [reply pytorch/matmul.py]:
```
    "8-bit integer (unsigned)": {
        "dtype": torch.uint8,
    },
    "8-bit integer (signed)": {
        "dtype": torch.int8,
    },
    "16-bit integer (signed)": {
        "dtype": torch.int16 or torch.short,
    },
    "64-bit integer (signed)": {
        "dtype": torch.int64 or torch.long,
    },
```

Comment (pacospace) [reply pytorch/matmul.py]: These ones could change to `uint8`, `int8`, `int16` and `int64`

File: pytorch/matmul.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import json
+from timeit import time
+
+import torch
+
+_LOGGER = logging.getLogger(__name__)
+
+_ARGS_DTYPE_MAP = {
+    "float32": {
+        "dtype": torch.float32 or torch.double,
```
Comment (fridex) [inline pytorch/matmul.py]: Is there some special case for `or`?

File: pytorch/matmul.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    else:
+        device = torch.device('cpu')
+
+    matrix1 = torch.ones((n, n), dtype=_ARGS_DTYPE_MAP[_ARGS_DTYPE]["dtype"], device=device)
+    matrix2 = torch.ones((n, n), dtype=_ARGS_DTYPE_MAP[_ARGS_DTYPE]["dtype"], device=device)
+
+    times = []
+
+    for i in range(_ARGS_REPS):
+        start = time.time()
```
Comment (fridex) [inline pytorch/matmul.py]: Please use `time.monotonic`.

File: pytorch/matmul.py (hunk 4 above)
Comment (pacospace) [reply pytorch/matmul.py]: Not really :)

File: pytorch/matmul.py (hunk 5 above)
Comment (pacospace) [reply pytorch/matmul.py]: I will change it also in TF :)

**Review by fridex — APPROVED** [review]
Excellent, thanks! :+1:

**Pull request merged**